pkg/restore/item_action.go

```diff
@@ -34,7 +34,23 @@ type ItemAction interface {
 	// should be returned, along with a warning (which will be logged but will not prevent
 	// the item from being restored) or error (which will be logged and will prevent the item
 	// from being restored) if applicable.
-	Execute(obj runtime.Unstructured, restore *api.Restore) (res runtime.Unstructured, warning error, err error)
+	Execute(input *RestoreItemActionExecuteInput) (*RestoreItemActionExecuteOutput, error)
+}
+
+// RestoreItemActionExecuteInput is a struct passed to Execute function
+// It contains item being processed by actions, unchanged item taken from backup
+// and restore spec
+type RestoreItemActionExecuteInput struct {
+	Item           runtime.Unstructured
+	ItemFromBackup runtime.Unstructured
+	Restore        *api.Restore
+}
+
+// RestoreItemActionExecuteOutput is a struct returned from Execute function
```

Comment:
`RestoreItemActionExecuteOutput contains the output variables for the ItemAction's Execution function.`